torchgeo/datasets/utils.py

```diff
@@ -628,6 +627,18 @@ def unbind_samples(sample: dict[Any, Sequence[Any]]) -> list[dict[Any, Any]]:
     return _dict_list_to_list_dict(sample)
 
 
+def interpolate_samples(
```

Comment:
This is intended to be a transform that fills gap of irregularly sampled time series. To be determined. Other transform functions are possible if we apply them on the sample level, otherwise we also have to provide a `collate_fn` since samples from the `TimeWindowGeoDataset` are not guaranteed to have the same time dimension.